test/TestDataFactory.js

```diff
@@ -0,0 +1,36 @@
+// ## TestDataFactory functions
+// Emulate old, pre-RDF/JS output (plain strings instead of Terms) to make tests pass
+
+// ### Creates an IRI
+function namedNode(iri) {
+  return iri;
+}
+
+// ### Creates a blank node
+function blankNode(name) {
+  return '_:b' + name;
+}
+
+// ### Creates a literal
+function literal(value, tag) {
+  if (tag) {
+    if (tag.length < 10) // hacky "heuristic" for test purposes only
```

Comment:
This version of of `literal()` brought the number of test failures down to 10.

But there's something going on with the value escaping, that's why the output differs for lang-tagged and datatyped literals. Try to change any of those and the failure number will go up.

I'm guessing the grammar produces differently escaped literals in different places.